core/src/main/scala/ai/h2o/sparkling/backend/converters/SparkDataFrameConverter.scala

```diff
@@ -40,23 +40,37 @@ object SparkDataFrameConverter extends Logging {
     spark.baseRelationToDataFrame(relation)
   }
 
-  def toH2OFrame(hc: H2OContext, dataFrame: DataFrame, frameKeyName: Option[String]): H2OFrame = {
+  def toH2OFrame(
+      hc: H2OContext,
+      dataFrame: DataFrame,
+      frameKeyName: Option[String] = None,
+      featureColsForConstCheck: Option[Seq[String]] = None): H2OFrame = {
     val df = dataFrame.toDF() // Because of PySparkling, we can receive Dataset[Primitive] in this method, ensure that
     // we are dealing with Dataset[Row]
     val flatDataFrame = flattenDataFrame(df)
     val schema = flatDataFrame.schema
-    val rdd = flatDataFrame.rdd // materialized the data frame
+    val rdd = flatDataFrame.rdd.cache()
```

Comment:
Shouldn't we cache the rdd rather to a disk in case of internal backend , where h2o and spark executors share memory space?

Reply:
My thought was "at the stage before training we should have plenty of RAM in most cases, why not use that?" - what do you think?

Reply:
H2OFrame as such in DKV takes usually majority of the memory. There is a rule of thumb for the memory of H2O cluster. The memory size of H2O cluster should be at least 4-5x more than the size of the dataset represented as CSV. 

if you cache the source Spark DF then there will be less space for the final H2OFrame since you keep them in memory at the same time.

Reply:
> "at the stage before training we should have plenty of RAM in most cases

data import and transfer into H2O has much higher memory requirements than the training itself typically

Reply:
It is better to prefer stability over performance - I think caching on disk is better in this case

Reply:
I added caching to disk when h2o backend mode is internal and default cache spark when external